### DISCUSSION TITLE
[ZK Stack] - Eth sender is behind - block commited is smaller than most recent state keeper batch

### DISCUSSION THREAD

**Discussion by wjw12** [zkSync-Community-Hub/zksync-developers#250] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.74.0-nightly (5c6a7e71c 2023-08-20)

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I followed the [prover_fri guide](https://github.com/matter-labs/zksync-era/blob/main/prover/prover_fri/README.md) trying to enable prover by GPU. I've set `dummy_verifier` to false. All the server, gateway, witness generator, prover, witness vector generator run successfully without a problem. 
However, `zk status prover` never shows a block proving is complete. 

My hardware: NVIDIA A4000
Driver: NVIDIA-SMI 545.23.08 Driver Version: 545.23.08    CUDA Version: 12.3

### Expected Behavior

I expect to see at least 1 block verified from `zk status prover` result

### Current Behavior

`zk status prover` result

==== FRI Prover status ====
State keeper: First batch: 0, recent batch: 461
L1 state: block verified: 0, block committed: 460
Eth sender is behind - block commited 460 is smaller than most recent state keeper batch 461.

### Additional Details

`zksync_proof_fri_compressor` occasionally prints `ERROR vise_exporter::exporter: Error submitting metrics to Prometheus push gateway err=error trying to connect: tcp connect error: Connection refused (os error 111) endpoint=http://127.0.0.1:9091/` but I don't think this is important.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by shahar4** [comment]
Thanks for reaching out with this! Eth Sender being behind isn't necessarily an issue. Your proof system night still be chugging along. Some questions:
1. When did you start the system? How long has it been running?
2. Have you played with batch sealing time or at all with how frequently batches are created, or did you leave the defaults?
3. Does the server send batches (`commit blocks`) as soon as they are created, or does the lag ever exceed one batch?

**Comment by shahar4** [comment, marked as answer]
@wjw12 if you merge main, run `zk`, and then run the `zk status prover` command again, you will see more details about the status of the prover

**Comment by ternianitha** [comment]
Based on the provided information, it seems like the FRI Prover is not successfully verifying any blocks, despite the witness generation and other processes running without errors. Here are some steps you can take to troubleshoot the issue:

Check Prover Logs: Review the logs of the FRI Prover to see if there are any errors or warnings that might indicate why it's not verifying blocks.

Verify GPU Configuration: Ensure that your GPU (NVIDIA A4000) is properly configured and recognized by the system. Check for any driver issues or compatibility issues with CUDA.

Monitor Resource Usage: Monitor the resource usage of the FRI Prover process, including GPU utilization, memory usage, and CPU usage. Ensure that the prover is actually utilizing the GPU for verification tasks.

Network Connectivity: Ensure that the prover is able to connect to the necessary endpoints and services required for block verification. Check for any network issues or firewall restrictions that might be preventing communication.

Check Eth Sender: Investigate the status of the Ethereum sender component to ensure that it's functioning correctly and not causing any delays or errors in the block verification process.

Review Configuration: Double-check the configuration settings for the FRI Prover and related components to ensure that everything is correctly configured for GPU-based verification.

By systematically reviewing these aspects and addressing any issues or discrepancies, you should be able to diagnose and resolve the issue with the FRI Prover not verifying blocks as expected.

**Comment by franksofa** [comment]
It seems that the FRI prover is not completing block verification despite all components running successfully. Check if the prover is correctly configured to utilize the GPU, ensure CUDA is properly installed, and verify that the prover is receiving and processing tasks. Additionally, investigate any potential errors or warnings in the prover logs that might provide insights into why block verification is not progressing.

**Comment by jonnybutlein** [comment]
It seems that the FRI prover is not able to catch up with the latest blocks generated by the state keeper. This could be due to various reasons such as resource constraints, configuration issues, or bottlenecks in the prover setup.

To troubleshoot this issue, you may need to:

Ensure that the prover setup has sufficient resources, especially GPU resources, to handle the workload.
Check the configuration files for any misconfigurations that might be causing the prover to fall behind.
Monitor system resource usage to identify any bottlenecks that might be slowing down the prover.
Verify that the communication between different components (state keeper, prover, etc.) is functioning correctly.
Look for any error logs or warnings that might provide clues about the root cause of the issue.
Once you identify the specific cause, you can take appropriate actions to address it and ensure that the FRI prover can catch up with the latest blocks successfully.

**Comment by antoniostoric** [comment]
FRI Prover does not show any blocks verified despite successful setup and running of all related components. Eth sender is behind according to zk status prover.
 zk status prover shows no blocks verified.
 Occasional ERROR messages from zksync_proof_fri_compressor regarding metrics submission to Prometheus push gateway, but considered unrelated.

**Comment by erinemsyntracy** [comment]
The user is trying to enable the prover by GPU following the prover_fri guide, but the zk status prover command is not showing any blocks verified. The user's hardware is an NVIDIA A4000 with driver version 545.23.08 and CUDA version 12.3. The zksync_proof_fri_compressor occasionally prints an error related to exporting metrics to Prometheus push gateway, but it is not clear if this is related to the issue. Without further information or access to the code repository, it is difficult to diagnose the exact issue.
